### DISCUSSION TITLE
Intelligent Fan Air Cooling System (Project 161)

### DISCUSSION THREAD

**Discussion by robertogl** [mathworks/MATLAB-Simulink-Challenge-Project-Hub#17] [Collaborate]
Contribute to the discussion by asking and/or answering questions, commenting, or sharing your ideas for solutions to [project #161](https://github.com/mathworks/MathWorks-Excellence-in-Innovation/tree/main/projects/Intelligent%20Fan%20Air%20Cooling%20System)

---
*:robot: An AI bot may reply to your comments with suggestions and guidance. It will not solve the project for you. Your comments may be processed by a third-party LLM. To direct a question to a specific person, tag them with @username.*

**Comment by swastik1308** [comment]
Hi, I am a student at College of Engineering and Technology and interested in working and collaborating on this project.

**Reply by mw-agrace** [reply]
Hi Swastik: 

Congratulations! You are the first to sign up for this project! I see you are in India where I know it is getting very hot now so this is probably a good project for you!

This would be a good project to carry through to prototyping. You might think about using the [Simulink Support Package for Arduino](https://www.mathworks.com/hardware-support/arduino-simulink.html) and building a [fan system](https://www.mathworks.com/videos/using-matlab-and-arduino-for-motor-control-100737.html) or just a [controlled power outlet](https://www.circuitbasics.com/build-an-arduino-controlled-power-outlet/) on an existing fan.  Good luck and let me know if you have any questions!

Andy

**Comment by Wandrys-dev** [comment]
Hello everyone. I am a mathematician and currently finishing statistics at the Federal University of Goiás. This will be my first project and I hope to be able to collaborate effectively.

**Reply by mw-agrace** [reply]
Welcome Wandrys! If you are looking for more of a data science aspect to this project, looking at weather patterns and providing recommendations on which parts of the world this type of system would perform well and more economically, and over which time periods, than a traditional air conditioner would be an interesting angle. Of course, there are many other sub-project too!

**Comment by swastik1308** [comment]
Hello Everyone,
I was wondering a solution for the problem, and it's kind of hypothetical, so I am looking forward to your suggestions. What I have thought is to build a control system for a twin fan system, with opposite direction, or a single window fan with some sort of winter mode for it to act as a heater. Further, based on the sensors and the set point, we can add some sort of speed and direction control mechanism using PID controller.

Optional: The project may be further advanced by adding intelligence, considering datasets with environmental parameters, region, season, etc.

Hope that sounds a bit realistic. Please let me know your views on this.

**Reply by robertogl** [reply]
Hi Swastik,
Yes your approach for the control system sounds proper to me. I expect the modeling, data acquisition, and prediction parts of the project being more critical and time consuming.

**Reply by mw-agrace** [reply]
Swastik: The best likely control scheme is a simple bang-bang (on-off) controller. The beauty of physical modeling is that once you have the Simulink/Simscape model, you can easily test out various control schemes including a PID controller.  The complexity and sophistication from this project comes from trying maintain an average temperature as close to the ideal as possible. For example, you will want to keep the fan running in the morning even if it goes below the desired temperature, if you know it's going to be hot day and want to cool the house more than normal in anticipation of the hot temperatures to follow. 

In other words,  if you have an estimate of daily temperatures and how they vary during the day and have modeled the thermal inertias of the building, you can use simulation to test our various control schemes.

**Comment by huntermanana** [comment]
Hi, I am a student at Makerere University and am very much interested in working and collaborating on this project. 
I am 90 percent done with this project though I am now stuck. This is how I have modeled it.
1. I successfully modeled the house, with the house temperature and outside temperature as input to the house in simulink.
2. I modeled the fan system in simscape.
3. I modeled the control system to the fan system in simulink using a PID controller.

The idea is to use the difference between the set temperature and the house temperature to tune and control whether the fan system comes on to cool down the house temperature in case it's too high and vice versa. It should be able to control and maintain a certain temperature in the house.
This is where I am stuck, I don't know how to change the house temperature to a desired set temperature using the fan system. 

I appreciate all the help I will get.

**Reply by danup415** [reply]
Hello huntermanana , Can you share the details of how the project has been done by you ??
Thanks in advance

**Comment by mukeshnarayana24** [comment]
Hello, I am Mukesh Narayana from India. The description mentioned that by using Arduino we have to design a model, Instead of Arduino can I use ESP32 Microcontroller?

**Reply by robertogl** [reply]
Absolutely, as long as you use MATLAB or Simulink. The board is also supported.
Here a couple of links that can be useful:
https://www.mathworks.com/help/supportpkg/arduinoio/ug/configure-setup-for-esp32-hardware.html
https://www.mathworks.com/help/supportpkg/arduinoio/ug/esp32-board-tips.html

Cheers!
